src/Compilers/CSharp/Portable/Symbols/TupleTypeSymbol.cs

```diff
@@ -363,6 +379,47 @@ internal static int IsMemberNameReserved(string name)
         }
 
         /// <summary>
+        /// Lookup well-known member declaration in provided type.
+        ///
+        /// If a well-known member of a generic type instantiation is needed use this method to get the corresponding generic definition and
+        /// <see cref="MethodSymbol.AsMember"/> to construct an instantiation.
+        /// </summary>
+        /// <param name="type">Type that we'll try to find member in.</param>
+        /// <param name="relativeMember">A reference to a well-known member type descriptor. Note however that the type in that descriptor is ignored here.</param>
+        /// <param name="accessWithinOpt">The assembly referencing the type.</param>
+        internal static Symbol GetWellKnownMemberInType(NamedTypeSymbol type, WellKnownMember relativeMember, AssemblySymbol accessWithinOpt)
+        {
+            Debug.Assert(relativeMember >= 0 && relativeMember < WellKnownMember.Count);
+            Debug.Assert(type.IsDefinition);
+
+            MemberDescriptor relativeDescriptor = WellKnownMembers.GetDescriptor(relativeMember);
+            return CSharpCompilation.GetRuntimeMember(type, ref relativeDescriptor, CSharpCompilation.SpecialMembersSignatureComparer.Instance, accessWithinOpt: accessWithinOpt);
+        }
+
+        /// <summary>
+        /// Lookup well-known member declaration in provided type and reports diagnostics.
+        /// </summary>
+        internal static Symbol GetWellKnownMemberInType(NamedTypeSymbol type, WellKnownMember relativeMember, AssemblySymbol accessWithinOpt, DiagnosticBag diagnostics, SyntaxNode syntax)
+        {
+            Symbol member = GetWellKnownMemberInType(type, relativeMember, accessWithinOpt);
+
+            if ((object)member == null)
+            {
+                Binder.Error(diagnostics, ErrorCode.ERR_PredefinedTypeMemberNotFoundInAssembly, syntax, relativeMember.ToString(), type.ToDisplayString(), accessWithinOpt.ToDisplayString());
```

Comment:
We never convert diagnostics arguments to string explicitly. One of the reasons that they should be formatted according to the target culture the diagnostics is formatted with. So, if default format doesn't quite what we want, we have FormattedSymbol class for that.